### DISCUSSION TITLE
Optimizing Test Structure

### DISCUSSION THREAD

**Discussion by coolbeevip** [camel-ai/camel#1151] [Ideas]
Currently, the project's test cases are in the camel/test directory. Some tests need a Google API key or an OpenAI API key to run correctly.

If a contributor hasn't set up these keys in their forked repo, their tests will fail when submitting a PR. But these keys aren’t free.

Could we organize the test cases into two categories: unit_tests and integration_tests? It would look like this:

* camel/tests/unit_tests (for essential tests)
* camel/tests/integration_tests (for tests that connect to external APIs)

This would make it easier for contributors to submit PRs during the CI/CD process while still keeping the necessary integration tests functional after merging.

**Comment by Wendong-Fan** [comment]
Thanks @coolbeevip ! Sorry for the late reply, it's really a great idea, we need to do this!
